Dockerfile

```
@@ -0,0 +1,41 @@
+# Multi-stage build to download the just binary and create a minimal container
+FROM alpine:latest as downloader
+
+# Version will be passed from GitHub Actions during release builds
+ARG JUST_VERSION
+# Docker automatically provides these for multi-platform builds
+ARG TARGETARCH
+ARG TARGETVARIANT
+
+# Install tools needed to download and extract the binary
+RUN apk add --no-cache wget tar
+
+# Download the appropriate just binary for the target architecture
+# Map Docker's platform names to just's release naming convention
+RUN set -eux; \
+    case "$TARGETARCH" in \
+        "amd64") ARCH="x86_64-unknown-linux-musl" ;; \
+        "arm64") ARCH="aarch64-unknown-linux-musl" ;; \
+        "arm") \
+            # Handle ARM variants - v7 maps to armv7, others to generic arm
+            case "$TARGETVARIANT" in \
+                "v7") ARCH="armv7-unknown-linux-musleabihf" ;; \
+                *) ARCH="arm-unknown-linux-musleabihf" ;; \
+            esac ;; \
+        *) echo "Unsupported architecture: $TARGETARCH" && exit 1 ;; \
+    esac; \
+    wget -O just.tar.gz "https://github.com/casey/just/releases/download/${JUST_VERSION}/just-${JUST_VERSION}-${ARCH}.tar.gz"; \
+    tar -xzf just.tar.gz; \
+    chmod +x just
```

As explained in [my comment](https://github.com/casey/just/issues/1497#issuecomment-1442903322) of the related issue, you can just pipe the output through into `tar` and extract the contents, it is already executable (_archive preserved it from build output_):

```suggestion
    wget -O - https://github.com/casey/just/releases/download/${JUST_VERSION}/just-${JUST_VERSION}-${ARCH}.tar.gz \
        | tar --directory /usr/local/bin --extract --gzip --file - --no-same-owner just
```

---

yeah i saw that; my containers don't always have internet access or limited bandwidth (ex: while travelling on planes).  docker provides a nice prepopulated caching layer for my dev-x.  i also add another logging wrapper around it to keep track of which commands were called and whatnot; so just isn't even installed - this is for validating alignment (tracking if an llm did a specific just command & the outcome) .. i wrote just-mcp; i use it in all sorts of weird ways. ;-)